backend/src/main/java/fr/zelytra/session/SessionSocket.java

```diff
@@ -172,32 +167,29 @@ private void handleLeaveMessage(Player player) {
 
     @OnClose
     public void onClose(Session session) {
-
-        // Clean up resources related to the session
-        sessionTimeoutTasks.remove(session.getId());
-
-        SessionManager manager = SessionManager.getInstance();
-        Player player = manager.getPlayerFromSessionId(session.getId());
-
-        if (player != null) {
-            manager.leaveSession(player);
-            Log.info("[" + player.getUsername() + "] Disconnected");
-            return;
-        }
-        Log.warn("[UNDEFINED PLAYER] Disconnected");
+        handleCloseSession(session);
     }
```

Comment:
in which conditions does onclose is called ? can it loop on itself ? seems weird